tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py

```diff
@@ -0,0 +1,231 @@
+
+        mounting_dir = "".join(
+            random.choice(string.ascii_lowercase + string.digits)
+            for _ in list(range(10))
+        )
+        log.info("Mount 1 subvolume on kernel and 1 subvolume on Fuse → Client1")
+
+        kernel_mounting_dir_1 = f"/mnt/cephfs_kernel{mounting_dir}_1/"
+        mon_node_ips = fs_util_v1.get_mon_node_ips()
+        subvol_path_kernel, rc = clients[0].exec_command(
+            sudo=True,
+            cmd=f"ceph fs subvolume getpath {default_fs} {subvolume_name}_1 {subvolume_group_name}",
+        )
+        fs_util_v1.kernel_mount(
+            [clients[0]],
+            kernel_mounting_dir_1,
+            ",".join(mon_node_ips),
+            sub_dir=f"{subvol_path_kernel.strip()}",
+            extra_params=f",fs={default_fs}",
+        )
+
+        fuse_mounting_dir_1 = f"/mnt/cephfs_fuse{mounting_dir}_1/"
+        subvol_path_fuse, rc = clients[0].exec_command(
+            sudo=True,
+            cmd=f"ceph fs subvolume getpath {default_fs} {subvolume_name}_2 {subvolume_group_name}",
+        )
+        fs_util_v1.fuse_mount(
+            [clients[0]],
+            fuse_mounting_dir_1,
+            extra_params=f" -r {subvol_path_fuse.strip()} --client_fs {default_fs}",
+        )
+
+        file_paths = [
+            f"/mnt/cephfs_kernel{mounting_dir}_1/large_file_kernel_1.txt",
+            f"/mnt/cephfs_fuse{mounting_dir}_1/large_file_fuse_1.txt",
+        ]
+
+        create_large_file_script = "create_large_file.py"
+        clients[0].upload_file(
+            sudo=True,
+            src="tests/cephfs/cephfs_bugs/create_large_file.py",
+            dst=f"/root/{create_large_file_script}",
+        )
+        clients[0].exec_command(
+            sudo=True,
+            cmd=f"python /root/{create_large_file_script} {' '.join(file_paths)}",
+            timeout=7200,
+        )
+
+        log.info(
+            "Check for the Ceph Health to see if there are any deadlock bw unlink and rename."
+        )
+        fs_util_v1.get_ceph_health_status(clients[0])
+
+        fsid = fs_util_v1.get_fsid(clients[0])
+        mds = fs_util_v1.get_active_mdss(clients[0], default_fs)
+        active_mdss_hostnames = [i.split(".")[1] for i in mds]
+        log.info(f"Active MDS Nodes are : {mds}")
+        for mds_node in mds_nodes:
+            if mds_node.node.hostname in active_mdss_hostnames:
+                log.info("Verify if there are repeated Revoke entries in MDS log.")
+                out, rc = mds_node.exec_command(
+                    sudo=True,
+                    cmd=f"grep -i 'mclientcaps(revoke)'  /var/log/ceph/{fsid}/ceph-mds*",
+                    check_ec=False,
+                )
+                log.info(out)
+                if "mclientcaps(revoke)" not in out:
+                    print(
+                        "Test is passed: No repeated mclientcaps Revoke entries found in MDS log."
+                    )
+                else:
+                    log.error(
```

Comment:
I have seen that with log.error executed, Test case is not marked Failed as return 0 happens.
with "mclientcaps(revoke)" in out, in else block we should add return 1 or we can raise an exception?

Reply:
Added CommandFailed exception